### PR TITLE
fix(dropdown-menu): fix positioning for responsive dropdown menu

### PR DESCRIPTION
**Pull request by ivanthai**
### Overview
There are a few bug fixes here related to a few bugs we encountered with DropdownMenu for full-screen which includes

- Position menu correctly when in full-screen - Added an additional css class and JS logic to properly position the menu above the bottom viewport
- Remove unneccesary scrollbar

### Demo
**Bad menu positioning and jerking motion**

buie-dropdown-menu-rpositioning-bug

**Fix**

buie-jarring-menu-bug-fix

### CODE REVIEW

File: src/components/dropdown-menu/DropdownMenu.scss [hunk 1]
```diff
@@ -54,13 +54,15 @@
                 box-shadow: $bdl-header-box-shadow;
             }
 
-            .submenu {
+            .submenu:not(.is-hidden) {
```
Comment (tjuanitas) [inline src/components/dropdown-menu/DropdownMenu.scss:57]: what happens if we remove `:not(.is-hidden)`?

I'm wondering if the extra specificity here will take precedence: https://github.com/box/box-ui-elements/blob/00fc868b7f67566927eaaa11a6c5f7eb24438961/src/components/menu/SubmenuItem.scss#L20-L23

File: src/components/dropdown-menu/DropdownMenu.scss [hunk 2]
```diff
@@ -36,7 +36,7 @@
         .aria-menu:not(.submenu) {
             height: 100%;
             margin-top: $bdl-header-height;
-            overflow: scroll;
+            overflow-x: auto;
```
[the remark this answers is not in the document]
Comment (ivanthai) [reply src/components/dropdown-menu/DropdownMenu.scss:39]: There should not be a case for horizontal scrolling for a full screen menu. I think I want the scrollbar to appear when there are more elements that can fit on screen

File: src/components/menu/SubmenuItem.tsx [hunk 3]
```diff
@@ -82,10 +83,12 @@ class SubmenuItem extends React.Component<SubmenuItemProps, SubmenuItemState> {
             submenuTriggerElBounding.right + submenuElBounding.width > rightBoundaryElementBounding.right;
         const isBottomAligned =
             submenuTriggerElBounding.top + submenuElBounding.height > bottomBoundaryElementBounding.bottom;
-
+        const isInlineBottom =
```
[the remark this answers is not in the document]
Comment (ivanthai) [reply src/components/menu/SubmenuItem.tsx:86]: it is hovering over the menu and is aligned at the bottom. Maybe a better name would be inside bottom aligned?

File: src/components/dropdown-menu/DropdownMenu.scss (hunk 1 above)
Comment (ivanthai) [reply src/components/dropdown-menu/DropdownMenu.scss:57]: It doesn't take precedence

Comment (ivanthai) [reply src/components/dropdown-menu/DropdownMenu.scss:57]: These styles only really apply when the submenu is not hidden anyways and it was easier to just omit is-hidden to not have to deal with style overrides

File: src/components/menu/SubmenuItem.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ivanthai) [reply src/components/menu/SubmenuItem.tsx:86]: I renamed it right-bottom-aligned

**Review by tjuanitas — APPROVED** [review]
looks good to me but had a final clarification on the name

File: src/components/dropdown-menu/DropdownMenu.scss [hunk 4]
```diff
@@ -54,13 +54,15 @@
...
                 top: 100%;
                 right: 0;
+                bottom: auto;
                 left: auto;
 
-                &.is-hidden {
-                    display: none;
+                &.is-right-bottom-aligned {
+                    top: auto;
+                    bottom: 100%;
```
Comment (tjuanitas) [inline src/components/dropdown-menu/DropdownMenu.scss:65]: I'm curious why `isRightBottomAligned` a better name when there doesn't seem to be any styles related to the horizontal position?

Comment (ivanthai) [reply src/components/dropdown-menu/DropdownMenu.scss:65]: This actually depends on the outer style

**Review by Machoper — APPROVED** [review]

**Pull request merged**